clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -163,6 +163,15 @@ else if (maxTimestamp == LogEntry.NO_TIMESTAMP)
             return new RecordsInfo(maxTimestamp, compressionType == CompressionType.NONE ? offsetOfMaxTimestamp : lastOffset);
     }
 
+    /**
+     * We set the sequence of the batch just before it is about to be sent out.
+     * @param baseSequence
+     */
+    public void closeWithSequence(int baseSequence) {
+        this.baseSequence = baseSequence;
```

Comment:
I see. I think the semantics of `closeWithSequence` do merit an exception if the batch is already closed, since the new sequence number will be ignored. Will update.